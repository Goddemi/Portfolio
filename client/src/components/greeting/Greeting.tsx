import React from "react";
import styled from "styled-components";

const Greeting = () => {
  return (
    <Container>
      <ProfileImg src={process.env.PUBLIC_URL + "/reserve/imgs/profile.jpeg"} />
      <SayHello>
        <MyName>침착한 프론트엔드 개발자 이강철 입니다.</MyName>
        <KeySentence>
          <span className="colored">직관적인 것</span>. 그리고{" "}
          <span className="colored">클린코드</span>를 추구합니다.
          <span className="blinking">|</span>
        </KeySentence>
      </SayHello>
    </Container>
  );
};

export default Greeting;

const Container = styled.div`
  display: flex;
  margin-top: 30px;
`;

const ProfileImg = styled.img`
  width: 380px;
  border-radius: ${(props) => props.theme.radius};
`;

const SayHello = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;

  span {
    .colored {
      color: ${(props) => props.theme.purple};
    }
  }

  .blinking {
    animation: blink-effect 1s step-end infinite;
  }

  @keyframes blink-effect {
    50% {
      opacity: 0;
    }
  }
`;

const MyName = styled.span`
  display: block;
  margin: 30px;
  font-size: 35px;
`;

const KeySentence = styled.span`
  margin: 30px;
  font-size: 35px;
`;
