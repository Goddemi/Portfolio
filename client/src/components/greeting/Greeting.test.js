import Greeting from "./Greeting";
import renderer from "react-test-renderer";

describe("Greeting", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Greeting />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
