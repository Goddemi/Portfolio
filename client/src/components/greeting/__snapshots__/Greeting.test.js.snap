// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting renders correctly 1`] = `
<div
  className="sc-bcXHqe eOIHv"
>
  <img
    className="sc-gswNZR fTpcmO"
    src="/reserve/imgs/profile.jpeg"
  />
  <div
    className="sc-dkrFOg kWFoiU"
  >
    <span
      className="sc-hLBbgP hxOekX"
    >
      침착한 프론트엔드 개발자 이강철 입니다.
    </span>
    <span
      className="sc-eDvSVe SxZcT"
    >
      <span
        className="colored"
      >
        직관적인 것
      </span>
      . 그리고
       
      <span
        className="colored"
      >
        클린코드
      </span>
      를 추구합니다.
      <span
        className="blinking"
      >
        |
      </span>
    </span>
  </div>
</div>
`;
